### PR TITLE
Validate binaries downloaded from GCS:

### LINKED ISSUE #10021: Validate downloaded binaries
Today during short hands on session we run into a problem which looked like corrupted Kubelet archive. Do we validate those when we upload them to salt master/download it through salt (or does the salt does this for us)?

### PR DESCRIPTION
**Pull request by zmerlynn**
- Set SHA1 for Kubernetes server binary and Salt tar in kube-env.
- Check SHA1 in configure-vm.sh. If the env variable isn't available,
  download the SHA1 from GCS and double check that.
- Fixes a bug in the devel path where we were actually uploading the
  wrong sha1 to the bucket.

Fixes #10021

### CODE REVIEW

**Comment by zmerlynn** [comment]
cc @roberthbailey @gmarek

File: cluster/gce/configure-vm.sh [hunk 1]
```diff
@@ -107,11 +107,23 @@ download-or-bust() {
...
+    echo "Failed to download file (${url}). Retrying."
   done
 }
 
+validate-hash() {
+  local -r file="$1"
+  local -r expected="$2"
+  local actual
+
+  actual=$(sha1sum ${file} | awk '{ print $1 }')
```
Comment (roberthbailey) [inline cluster/gce/configure-vm.sh]: do you need an || true here?

File: cluster/gce/configure-vm.sh [hunk 2]
```diff
@@ -430,31 +442,46 @@ EOF
   fi
 }
 
+function try-download-release() {
+  # TODO(zmerlynn): Now we REALLy have no excuse not to do the reboot
+  # optimization.
+
+  # TODO(zmerlynn): This may not be set yet by everyone (GKE).
```
Comment (roberthbailey) [inline cluster/gce/configure-vm.sh]: Will this case will also be hit when launching released versions (e.g. running `get.k8s.io | bash`)?

Comment (zmerlynn) [inline cluster/gce/configure-vm.sh]: No, that's yet another path. Fix incoming.

Comment (zmerlynn) [inline cluster/gce/configure-vm.sh]: Correction: I think this pull works for that script. What it doesn't do is handle the upgrade/pull case correctly (the `common.sh` path).

**Comment by zmerlynn** [comment]
PTAL

**Comment by roberthbailey** [comment]
LGTM.

**Pull request merged**